File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EnumNameHasEnumSuffix.cs
namespace Tests.Diagnostics
{
    public enum MyEnum //Noncompliant {{Rename this enumeration to remove the 'Enum' suffix.}}
//              ^^^^^^
    {
        Value
    }
    public enum MyFlags //Noncompliant {{Rename this enumeration to remove the 'Flags' suffix.}}
    {
        Value
    }
    public enum MyEnum2
    {
        Value
    }
    public class ClassEnum
    {
    }
}

File: analyzers/tests/SonarAnalyzer.Rules.Tests/Cases/EnumNameHasEnumSuffix.vb
Namespace Tests.Diagnostics
    Public Enum MyEnum ' Noncompliant {{Rename this enumeration to remove the 'Enum' suffix.}}
'               ^^^^^^
        Value
    End Enum
    Public Enum MyFlags ' Noncompliant
        Value
    End Enum
    Public Enum MyEnum2
        Value
    End Enum
    Public Class ClassEnum
    End Class
End Namespace